### PR TITLE
ts_web_test & karma fixes for Windows

### PR DESCRIPTION
**Pull request by gregmagolan**
ts_web_test and karma fixes for Windows

/cc @alexeagle

### CODE REVIEW

**Comment by gregmagolan** [comment]
Also fixes #159

File: internal/karma/karma.conf.js [hunk 1]
```diff
@@ -1,5 +1,56 @@
...
+// environment variable. This line points Karma to use puppeteer instead.
+// See
+// https://github.com/karma-runner/karma-chrome-launcher/blob/master/README.md#headless-chromium-with-puppeteer
+process.env.CHROME_BIN = require('puppeteer').executablePath();
+
+let files = [
+  TMPL_files
+];
+
+const manifestFile = path.join(process.env.TEST_SRCDIR, 'MANIFEST');
```
Comment (alexeagle) [inline internal/karma/karma.conf.js]: could you add docs on what the code below does?

File: internal/karma/karma.conf.js [hunk 2]
```diff
@@ -1,5 +1,56 @@
...
+if (fs.existsSync(manifestFile)) {
```
Comment (alexeagle) [inline internal/karma/karma.conf.js]: if for some misconfiguration this file is missing, will we just silently fail in an odd way? would it be better to rather fail early?

File: internal/karma/karma.conf.js [hunk 3]
```diff
@@ -1,5 +1,56 @@
...
+  const manifest = {};
```
Comment (alexeagle) [inline internal/karma/karma.conf.js]: what's in this struct? what's in the manifest? docs?

Comment (alexeagle) [inline internal/karma/karma.conf.js]: what's in this struct? what's in the manifest? docs?

File: internal/karma/karma.conf.js [hunk 4]
```diff
@@ -1,5 +1,56 @@
...
+  for (l of fs.readFileSync(manifestFile, 'utf8').split('\n')) {
+    const m = l.split(' ');
+    manifest[m[0]] = m[1];
+  }
+  const manifestKeys = Object.keys(manifest);
+  files = files.map((f) => {
+    if (manifestKeys.includes(f)) {
```
Comment (alexeagle) [inline internal/karma/karma.conf.js]: this is O(n^2), but should be easy to avoid using a map

File: internal/karma/karma.conf.js (hunk 2 above)
Comment (gregmagolan) [reply internal/karma/karma.conf.js]: The file is expected to be missing in linux and OSX. It should only be present on Windows.

File: internal/karma/karma.conf.js (hunk 3 above)
Comment (gregmagolan) [reply internal/karma/karma.conf.js]: I'll add some comments. I couldn't find anything on the MANIFEST file in the Bazel docs.

File: internal/karma/karma.conf.js (hunk 1 above)
Comment (gregmagolan) [reply internal/karma/karma.conf.js]: 👍

File: internal/karma/karma.conf.js (hunk 4 above)
Comment (gregmagolan) [reply internal/karma/karma.conf.js]: Thanks. Updated to remove the double lookup. It looks like O(n*m) to me where n is the number of test files and m is the number of files in the MANIFEST.

**Comment by gregmagolan** [comment]
@mprobst Please review when you have some time

**Pull request closed without merging**